content/automate/ManageIQ/System/Event/StateMachines/Refresh.class/__methods__/check_refreshed.rb

```diff
@@ -0,0 +1,63 @@
+#
+# Description: This method checks the refresh completion.
+#
+
+module ManageIQ
+  module Automate
+    module System
+      module Event
+        module StateMachines
+          module Refresh
+            class CheckRefreshed
+              STATE_FINISHED = 'Finished'.freeze
+
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                check_status(refresh_task)
+              end
+
+              private
+
+              def check_status(task)
+                case task.state
+                when STATE_FINISHED
+                  @handle.root['ae_result'] = 'ok'
+                else
+                  @handle.root['ae_result'] = 'retry'
+                  @handle.root['ae_retry_interval'] = 1.minute
+                end
+              end
+
+              def refresh_task
+                task_id = @handle.get_state_var(:refresh_task_id).try(:first)
+                @handle.log(:info, "Stored refresh task ID: [#{task_id}]")
+                if task_id.nil?
+                  @handle.log(:error, 'Refresh task id not found')
+                  raise 'Refresh task id not found'
+                end
+                fetch_task(task_id)
+              end
+
+              def fetch_task(task_id)
+                @handle.object.attributes.sort.each { |k, v| @handle.log(:info, "#{k}: #{v}") }
```

Comment:
@lfu 
Is this just for logging purpose? Do we need it?

Reply:
@mkanoor This is for information purpose. It would be useful if anything goes wrong. I would prefer to print it out for our good.

Reply:
We can get this information from the automation URL in the log already (in a less user-friendly format).  This adds more clutter to an already verbose log.  I would advise dropping it.